Update project build dependencies

friederbluemle opened the pull request:
Nice project! :+1:

I had some trouble locally compiling the app, so I updated and cleaned up some of the build scripts. It's now using the latest version of the Android Gradle plugin (`3.6.2`), as well as a matching Gradle Wrapper (`5.6.4`).

Please check individual commits, and let me know if anything is unclear.

Supersedes and closes #163, closes #164, closes #166, closes #167

Also updates GitHub Actions CI to also run for pull requests:

![ci][1]

<sub><sup>Once approved, please use a *normal* GitHub merge (i.e. **NO rebase/squash** merge) to integrate the commit(s) from the PR head branch. The changes are broken up into meaningful, [atomic commits](https://www.freshconsulting.com/atomic-commits/), and my branch should already be up-to-date with the latest base branch. If it isn't, or if you want me to change anything, please let me know, and I will update the branch as soon as possible. Thank you!</sup></sub>

[1]: https://github.com/friederbluemle/Android-Disassembler/workflows/ci/badge.svg

KYHSGeekCode commented on .gitignore:
```diff
@@ -1,11 +1,9 @@
-app/.cxx
-app/.externalNativeBuild
 local.properties
-*.hprof
```
I think I had added this line because the file was pushed after I tried profiling. Is there any reason for deleting this line? (.hprof)

friederbluemle replied:
Good point, I'll add it back.

KYHSGeekCode commented on app/build.gradle:
```diff
@@ -108,8 +108,7 @@ dependencies {
     implementation 'androidx.preference:preference:1.1.0'
-    implementation "com.mikepenz:aboutlibraries-core:v8.0.0-rc03"
-    implementation "com.mikepenz:aboutlibraries:7.1.0"
+    implementation "com.mikepenz:aboutlibraries:8.1.0"
```
If this commit's build successes, I will definitely agree with your commit. Well, nice! https://github.com/mikepenz/AboutLibraries/commit/01350cb43c4169ab5e01d4ab9507dfdd3363e376

friederbluemle replied:
Yes, there were some issue with `aboutlibraries`, but I was able to figure it out (basically, it required the new declaration in `AndroidManifest.xml`).

KYHSGeekCode commented on gradlew:
```diff
@@ -28,7 +44,7 @@ APP_NAME="Gradle"
 APP_BASE_NAME=`basename "$0"`
 
 # Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-DEFAULT_JVM_OPTS=""
```
What is the purpose of this line?

friederbluemle replied:
This is the auto-generated `gradlew` script by the Gradle `wrapper` task.

KYHSGeekCode commented:
I've set up four checks in my repository like below.
![image](https://user-images.githubusercontent.com/35697677/78199791-9f691780-74c7-11ea-8c30-0fe382fa7bcc.png)

Do you know why only two checks are being run on this pull request?
![image](https://user-images.githubusercontent.com/35697677/78199854-d0494c80-74c7-11ea-8735-fd02456aa0b6.png)

KYHSGeekCode commented:
By the way, I checked your branch and found that GitHub CI works fine, too :)

friederbluemle commented:
Hi @KYHSGeekCode - Thanks for your comments. I addressed them and updated the PR.

The reason GitHub Actions CI was not running for PRs was a missing `pull_request` trigger. I also updated the ci script and added it as a separate commit.

Now you will also have a CI status badge in README.md :+1:
